examples/gno.land/r/matijamarjanovic/home/config.gno

```diff
@@ -0,0 +1,70 @@
+package home
+
+import (
+	"errors"
+	"std"
+)
+
+var (
+	mainAddr   std.Address // tony's main address
+	backupAddr std.Address // backup address
+
+	errorInvalidAddr string
+	errorUnauthorized string
+)
+
+func init() {
+	mainAddr = "g1ej0qca5ptsw9kfr64ey8jvfy9eacga6mpj2z0y"
+	errorInvalidAddr = "config: invalid address"
+	errorUnauthorized = "config: unauthorized"
+}
+
+func Address() std.Address {
+	return mainAddr
+}
+
+func Backup() std.Address {
+	return backupAddr
+}
+
+func SetAddress(newAddress std.Address) error {
+	if !newAddress.IsValid() {
+		return errors.New(errorInvalidAddr)
+	}
+
+	if err := checkAuthorized(); err != nil {
+		return err
+	}
+
+	mainAddr= newAddress
+	return nil
+}
+
+func SetBackup(newAddress std.Address) error {
+	if !newAddress.IsValid() {
+		return errors.New(errorInvalidAddr)
+	}
+
+	if err := checkAuthorized(); err != nil {
+		return err
+	}
+
+	backupAddr= newAddress
+	return nil
+}
+
+func checkAuthorized() error {
+	caller := std.PrevRealm().Addr()
+	if caller != mainAddr|| caller != backupAddr{
+		return errors.New(errorUnauthorized)
+	}
+
+	return nil
+}
+
+func AssertAuthorized() {
```

Comment:
These two functions Leon made so he could show that it can be done in two different ways, although in official documentation the use of panic is encouraged.

Reply:
With this I agree, at the time I was uncertain so I decided to lean on the code from the example. Getting fixed in the next commit.

Reply:
So reading this I almost got confused but I remembered the reason this is the way it is. Like I mentioned [here](https://github.com/gnolang/gno/pull/2916#discussion_r1799594051), there was a discussion on this topic that explains the difference between these functions, there is also a solution to the dilemma included in the discussion, but as this code works as intended for my homepage I decided to not change it and leave it as is - the thing is both my functions used the std.GetPrevRealm().Addr() to make the caller which wouldn't work if the config.gno and home.gno were not in the same package. The way it would work is like it is now - checkAuthorized uses std.GetOrigCaller() and AssertAuthorized uses std.GetPrevRealm().Addr().